docs/user/external-database.md

```diff
@@ -0,0 +1,256 @@
+# Configuring External PostgreSQL Database
+
+FlightCtl supports using an external PostgreSQL database instead of the built-in database deployment. This is useful for large enterprise customers often have strict security policies covering all database components, production environments where you want to manage your database separately or use a managed database service.
```

Comment:
```suggestion
Flight Control supports using an external PostgreSQL database instead of the built-in database deployment. This is useful for using an existing database, implementing specific security policies or high availability, or using a managed database service.
```

Reply:
Replace flightctl with Flight Control for all intances when refering to the project, unless it's the CLI or service like flightctl-api

Reply:
> Replace flightctl with Flight Control 

replaced

> We usually have "enabled: ". Would it make sense to use this, and then if the db isn't enabled then we know it's external?

`enabled: false` doesn't not associates with `external` straightforward. Also in both cases the `db:` part is required, which makes `enabled: false` even more problematic. I see this `external` as a flag, the same as `type`. Maybe to change it to `internal` and to make it true/false.

> Doesn't this just call ./deploy/scripts/wait-for-database.sh, which tries to connect to the DB?
1. As discussed on the design meeting, we're expecting this DB to exist and to be ready, so there is no need to wait for it to be ready.
2. This script is in a `Containerfile.db-setup`, which doesn't run for external DB.